tests/link/jax/test_as_jax_op.py

```diff
@@ -0,0 +1,474 @@
+import jax
+import jax.numpy as jnp
+import numpy as np
+import pytest
+
+import pytensor.tensor as pt
+from pytensor import as_jax_op, config, grad
+from pytensor.graph.fg import FunctionGraph
+from pytensor.link.jax.ops import JAXOp
+from pytensor.scalar import all_types
+from pytensor.tensor import TensorType, tensor
+from tests.link.jax.test_basic import compare_jax_and_py
+
+
+def test_two_inputs_single_output():
+    rng = np.random.default_rng(1)
+    x = tensor("x", shape=(2,))
+    y = tensor("y", shape=(2,))
+    test_values = [
+        rng.normal(size=(inp.type.shape)).astype(config.floatX) for inp in (x, y)
+    ]
+
+    def f(x, y):
+        return jax.nn.sigmoid(x + y)
+
+    # Test with as_jax_op decorator
+    out = as_jax_op(f)(x, y)
+    grad_out = grad(pt.sum(out), [x, y])
+
+    fg = FunctionGraph([x, y], [out, *grad_out])
+    fn, _ = compare_jax_and_py(fg, test_values)
+    with jax.disable_jit():
+        fn, _ = compare_jax_and_py(fg, test_values)
+
+    # Test direct JAXOp usage
+    jax_op = JAXOp(
+        [x.type, y.type],
+        [TensorType(config.floatX, shape=(2,))],
+        f,
+    )
+    out = jax_op(x, y)
+    grad_out = grad(pt.sum(out), [x, y])
+    fg = FunctionGraph([x, y], [out, *grad_out])
+    fn, _ = compare_jax_and_py(fg, test_values)
+
+
+def test_two_inputs_tuple_output():
+    rng = np.random.default_rng(2)
+    x = tensor("x", shape=(2,))
+    y = tensor("y", shape=(2,))
+    test_values = [
+        rng.normal(size=(inp.type.shape)).astype(config.floatX) for inp in (x, y)
+    ]
+
+    def f(x, y):
+        return jax.nn.sigmoid(x + y), y * 2
+
+    # Test with as_jax_op decorator
+    out1, out2 = as_jax_op(f)(x, y)
+    grad_out = grad(pt.sum(out1 + out2), [x, y])
+
+    fg = FunctionGraph([x, y], [out1, out2, *grad_out])
+    fn, _ = compare_jax_and_py(fg, test_values)
+    with jax.disable_jit():
+        # must_be_device_array is False, because the with disabled jit compilation,
+        # inputs are not automatically transformed to jax.Array anymore
+        fn, _ = compare_jax_and_py(fg, test_values, must_be_device_array=False)
+
+    # Test direct JAXOp usage
+    jax_op = JAXOp(
+        [x.type, y.type],
+        [TensorType(config.floatX, shape=(2,)), TensorType(config.floatX, shape=(2,))],
+        f,
+    )
+    out1, out2 = jax_op(x, y)
+    grad_out = grad(pt.sum(out1 + out2), [x, y])
+    fg = FunctionGraph([x, y], [out1, out2, *grad_out])
+    fn, _ = compare_jax_and_py(fg, test_values)
+
+
+def test_two_inputs_list_output_one_unused_output():
```

Comment:
Merge this with the previous tests. Here is how we did it recently to test the grad of multi-output Ops: https://github.com/pymc-devs/pytensor/blob/19dafe42e2b34037b0f906d61116fa0d1de5025c/tests/tensor/test_nlinalg.py#L260-L278

You don't need to use verif_grad, since you're using compa_jax_and_py. The point is you can use `pytest.parametrize` and save a lot of lines since the tests are mostly identical?